R/ignore.R

```diff
@@ -22,5 +22,14 @@ use_build_ignore <- function(files, escape = TRUE, base_path = ".") {
 escape_path <- function(x) {
   x <- gsub("\\.", "\\\\.", x)
   x <- gsub("/$", "", x)
+
+  # at this point the string replacement for an absolute windows path would
+  # approximately be: C:\Users\hadley\...
+  # this causes the PCRE regex compilation error, which wants
+  # C:\\Users\hadley\...
+  #thus this regex will replace, if it detects a windows absolute path delineated
+  # by <Driveletter>:\ and replace it with <Driveletter>:\\ such that the regex
+  # doesn't explode.
+  x <- gsub("(^\\D:)", "\\1\\\\", x)
```

Comment:
I think `[A-Z]` would be clearer than `\\D`, and I'd usually put the anchor outside of the parens.